bin/bash_stub.sh

```diff
@@ -55,8 +55,12 @@ function send-to-server {
   echo -n "${2}" | nc localhost ${1}
 }
 
-function extract-properties {
-  echo -n "${1}" | sed -En "s/^\"${2}\":\"?([^,\"]*)\"?,?$/\1/gp"
+function extract-number-properties {
+  echo -n "${1}" | sed -En "s/^\"${2}\":([0-9]+),?$/\1/gp"
+}
+
+function extract-string-properties {
+  echo -n "${1}" | sed -En "s/^\"${2}\":\"(.*)\",?$/\1/gp"
```

Comment:
It doesn't really matter much, but these two regexes are almost identical. I wonder if we should extract the core regex that they share

Reply:
I agree. I think the biggest problem with that is the weird way bash deals with the variables. That is, the common part includes interpolation. Could be easy, could be hard?!

Reply:
I'm waiting to hear back from our QA team :) If they think this is good, I'll merge and release just to get this nasty but out of there. Such a big miss :(

Reply:
It's nearly impossible to not miss something when dealing with regexes...